### ISSUE TITLE
GraphQL API v2: Expense amount is not using Amount type

### ISSUE THREAD

**Issue by znarf** [opencollective/opencollective#5198]
As a key pattern in GraphQL API v2, we want all amounts to be expressed using the dedicated `Amount` type.

Expense is currently using the older pattern (`amount` as `GraphQLInt`), separated `currency`.

We should also fix `ExpenseItemCreateInput` and use `AmountInput` instead 

As this is a breaking change, we need to have a proper release strategy.

**Comment by Betree** [comment]
@znarf Something to be used like this?

```gql
expense(id: 4242) {
  amountInExpenseCurrency: amount(currencySource: EXPENSE) {
    valueInCents
    currency
  }
  amountInAccountCurrency: amount(currencySource: ACCOUNT) {
    valueInCents
    currency
  }
  amountInHostCurrency: amount(currencySource: HOST) {
    valueInCents
    currency
  }
}
```

**Comment by znarf** [comment]
Yes, that's one way. Or people could eventually ask a conversion in a specific currency. We could have both.

```
amountInExpenseCurrency: amount(currency: $currency) {
    valueInCents
    currency
  }

variables: { currency: expense.currency }
```

**Comment by Betree** [comment]
> Yes, that's one way. Or people could eventually ask a conversion in a specific currency. We could have both.
> 
> ```
> amountInExpenseCurrency: amount(currency: $currency) {
>     valueInCents
>     currency
>   }
> 
> variables: { currency: expense.currency }
> ```

Yep, having both is interesting. The tricky part about specifying the exact currency is that you don't necessarily know the collective/host currencies before the initial load.

**Comment by Betree** [comment]
I'm going with the `amountV2` + `currencySource` proposal:

![image](https://user-images.githubusercontent.com/1556356/153407810-d7892fed-510e-4e8b-a623-83c64ea6f6b2.png)

**Comment by Betree** [comment]
> Great. What's the default if `currencySource` is not passed?

It defaults to `currencySource: EXPENSE`.

![image](https://user-images.githubusercontent.com/1556356/153410012-88e2d752-fe76-4808-8218-dc98c23717d4.png)


> So, we name it `amountV2` and we move it to `amount` as soon as possible (I think some external people are relying on it already, so we should try to communicate that breaking change in a proper way)

Yep. Until GraphQL introduces something like Unions, we're stuck with this 3 steps process:
1. Introduce new `*V2` field, incentive everyone to use it by deprecating the old one
2. Remove old field
3. Re-introduce old field with new type and incentive everyone to move to it
4. Remove `*V2`

It's great if we have a way to directly message users, so we could add something in the logs to see who's using them. An update on https://opencollective.com/engineering/updates would also be helpful. In all cases, we'll have to move slowly.

**Comment by Betree** [comment]
I'm actually tempted to move with another proposal that I feel is more aligned with the latest multi-currency expenses changes by adding a new `amounts` field looking like this:

```es6
expense = {
  amounts: {
    inExpenseCurrency: {
      amount: { valueInCents: 100, currency: 'USD' },
    },
    inAccountCurrency: {
      amount: { valueInCents: 110, currency: 'EUR' },
      fxRate: {
        source: 'WISE',
        value: 1.1,
        isApproximate: false, // Whether this is a rough estimate or the exact amount (i.e. from the quote)
        fromCurrency: 'USD',
        toCurrency: 'EUR',
        date: '2022-02-09T00:00:00,
      }
    },
    inHostCurrency: {
      amount: { valueInCents: 110, currency: 'NZD' },
      fxRate: {
        source: 'OPENCOLLECTIVE',
        value: 1.1,
        isApproximate: true, // Whether this is a rough estimate or the exact amount (i.e. from the quote)
        fromCurrency: 'USD',
        toCurrency: 'NZD',
        date: '2022-02-09T00:00:00,
      }
    },
  }
}
```

This will also unblock https://github.com/opencollective/opencollective/issues/5300 by providing more info on the FX rate source. Some usage examples:

```es6
expense.amounts.inExpenseCurrency.amount.value; // Expense amount in expense currency
expense.amounts.inCollectiveCurrency.amount.value; // Expense amount in collective currency
expense.amounts.inCollectiveCurrency.fxRate.value; // Expense to collective currency FX rate
```

Happy to hear feedback about this approach.

**Comment by Betree** [comment]
Alternative way to solve the need: keep the original plan with `amountV2`, and update `accountCurrencyFxRate` to return an object like defined above. We would also need to introduce a new `hostCurrencyFxRate`.

```es6
expense = {
  accountCurrencyFxRate: {
        source: 'WISE',
        value: 1.1,
        isApproximate: false, // Whether this is a rough estimate or the exact amount (i.e. from the quote)
        fromCurrency: 'USD',
        toCurrency: 'EUR',
        date: '2022-02-09T00:00:00,
  }
}
```

**Comment by Betree** [comment]
@znarf Good idea, then the long-term vistion would be something like:

```gql
expense {
  amountInAccountCurrency: amount(currencySource: ACCOUNT) {
    valueInCents
    currency
    fxRate {
      value
      fromCurrency
      source
      date
    }
  }
}
```

**Comment by Betree** [comment]
@znarf Implemented the aforementioned design in https://github.com/opencollective/opencollective-api/pull/7254
